jsaddle/js/jsaddle-core.js

```diff
@@ -176,33 +176,54 @@ function jsaddle(global, sendRsp, processSyncCommand, RESPONSE_BUFFER_MAX_SIZE)
     }
     return syncRequests.dequeue();
   };
+  var syncDepth = 0;
   var processAllEnqueuedReqs = function() {
     while(!syncRequests.isEmpty()) {
-      var req = syncRequests.dequeue();
-      if(!req.Right) {
-        throw "processAllEnqueuedReqs: req is not Right; this should never happen because Lefts should only be sent while a synchronous request is still in progress";
+      var tuple = syncRequests.dequeue();
+      var syncReq = tuple[1];
+      if(syncReq.tag !== 'Req') {
+        throw "processAllEnqueuedReqs: syncReq is not SyncBlockReq_Req; this should never happen because Result/Throw should only be sent while a synchronous request is still in progress";
+      }
+      if (tuple[0] > syncDepth) {
+        console.warn ("processAllEnqueuedReqs: queue contains a request for a frame which has exited");
```

Comment:
Did you consider making this fatal? We can keep going here, but am I correct in thinking something is really wrong if we get the wrong syncDepth?

Reply:
fixed this in #10